commit 651d8883a611d8e77f2b36f117b8ca05ba4be8ed
Author: Dave Stevenson <dave.stevenson@raspberrypi.org>
Date:   Mon Feb 13 13:11:41 2017 +0000

    BCM2835-V4L2: Correctly denote key frames in encoded data
    
    Forward MMAL key frame flags to the V4L2 buffers.
    
    Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>

diff --git a/drivers/media/platform/bcm2835/bcm2835-camera.c b/drivers/media/platform/bcm2835/bcm2835-camera.c
index e69731320f4e..6bdec0806126 100644
--- a/drivers/media/platform/bcm2835/bcm2835-camera.c
+++ b/drivers/media/platform/bcm2835/bcm2835-camera.c
@@ -413,6 +413,9 @@ static void buffer_cb(struct vchiq_mmal_instance *instance,
 			dev->capture.last_timestamp = buf->vb.vb2_buf.timestamp;
 
 			vb2_set_plane_payload(&buf->vb.vb2_buf, 0, length);
+			if (mmal_flags & MMAL_BUFFER_HEADER_FLAG_KEYFRAME)
+				buf->vb.flags |= V4L2_BUF_FLAG_KEYFRAME;
+
 			v4l2_dbg(1, bcm2835_v4l2_debug, &dev->v4l2_dev,
 				"Buffer has ts %llu",
 				dev->capture.last_timestamp);
